influxdb3_write/src/write_buffer/validator.rs

```diff
@@ -682,119 +763,17 @@ fn convert_v3_parsed_line(
     );
 }
 
-impl<'lp> WriteValidator<LinesParsed<'lp, ParsedLine<'lp>>> {
-    /// Convert a set of valid parsed lines to a [`ValidatedLines`] which will
-    /// be buffered and written to the WAL, if configured.
-    ///
-    /// This involves splitting out the writes into different batches for each chunk, which will
-    /// map to the `Gen1Duration`. This function should be infallible, because
-    /// the schema for incoming writes has been fully validated.
-    pub(crate) fn convert_lines_to_buffer(
-        self,
-        ingest_time: Time,
-        gen1_duration: Gen1Duration,
-        precision: Precision,
-    ) -> ValidatedLines {
-        let mut table_chunks = HashMap::new();
-        let line_count = self.state.lines.len();
-        let mut field_count = 0;
-        let mut tag_count = 0;
-
-        for (line, _raw_line) in self.state.lines.into_iter() {
-            field_count += line.field_set.len();
-            tag_count += line.series.tag_set.as_ref().map(|t| t.len()).unwrap_or(0);
-
-            convert_v1_parsed_line(
-                Arc::clone(&self.state.db_schema),
-                line,
-                &mut table_chunks,
-                ingest_time,
-                gen1_duration,
-                precision,
-            );
-        }
-
-        let write_batch = WriteBatch::new(
-            self.state.catalog.db_schema.id,
-            Arc::clone(&self.state.catalog.db_schema.name),
-            table_chunks,
-        );
-
-        ValidatedLines {
-            line_count,
-            field_count,
-            index_count: tag_count,
-            errors: self.state.errors,
-            valid_data: write_batch,
-            catalog_updates: self.state.catalog_batch,
-        }
-    }
+struct QualifiedLine {
```

Comment:
I think with this change we could maybe use QualifiedLine as the thing that gets serialized in the WAL? Or get to a point where we don't need to do yet another conversion from the parsed and validated struct to whatever gets wal serialized. Something to consider for later.

Reply:
Yeah, it could probably just be the `Row` type used in the WAL. I will see if that is an easy change.

Reply:
I'll do this in another PR since this one is already massive, but I don't think it is hard. I opened https://github.com/influxdata/influxdb/issues/25512